bin/impl/redeploy.sh

```diff
@@ -15,7 +15,18 @@
 # limitations under the License.
 
 # stop if any command fails
-set -e  
+set -e
+
+function verify_checksum() {
+  tarball=$1
+  expected_md5=$2
```

Comment:
could have a sanity check on expected_md5... ensure its of correct len and has expected chars....